dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/io/readers/SampleReader.java

```diff
@@ -0,0 +1,187 @@
+                        ", batch.batch_id AS " + SampleRowMapper.BATCH_ID +
+                        ", batch.loc_batch_id AS " + SampleRowMapper.BATCH_NAME +
+                        ", indiv.loc_ind_id_upp AS " + SampleRowMapper.INDIVIDUAL_NAME +
+                        ", population.loc_pop_id AS " + SampleRowMapper.POPULATION +
+                        ", indiv.ind_id AS " + SampleRowMapper.INDIVIDUAL_ID +
+                        ", subind.submitted_ind_id AS " + SampleRowMapper.SUBMITTED_INDIVIDUAL_ID +
+                        ", ped.pa_ind_id AS " + SampleRowMapper.FATHER_ID +
+                        ", ped.ma_ind_id AS " + SampleRowMapper.MOTHER_ID +
+                        ", ped.sex AS " + SampleRowMapper.SEX
+        );
+        factoryBean.setFromClause(
+                "FROM " +
+                        "subind " +
+                        "JOIN submittedindividual indiv on indiv.submitted_ind_id = subind.submitted_ind_id " +
+                        "JOIN pedigreeindividual ped on indiv.ind_id = ped.ind_id " +
+                        "JOIN batch on subind.batch_id = batch.batch_id " +
+                        "JOIN population on population.pop_id = indiv.pop_id " +
+                        "JOIN subsnp sub ON subind.subsnp_id = sub.subsnp_id " +
+                        "JOIN snpsubsnplink link ON sub.subsnp_id = link.subsnp_id " +
+                        "JOIN b" + dbsnpRelease + "_snpcontigloc loc on loc.snp_id = link.snp_id " +
+                        "JOIN b" + dbsnpRelease + "_contiginfo ctg ON ctg.contig_gi = loc.ctg_id "
+        );
+        factoryBean.setWhereClause(
+                "WHERE " +
+                        "sub.subsnp_id = :ss_id " +
+                        "AND batch.batch_id = :batch " +
+                        "AND ctg.group_term IN (:assemblyType) " +
+                        "AND ctg.group_label LIKE :assembly "
+        );
+        factoryBean.setSortKey(SampleRowMapper.SUBMITTED_INDIVIDUAL_ID);
+
+        return factoryBean.getObject();
+    }
+
+    private Map<String, Object> getParametersMap(String dbsnpRelease, int batch, String assembly,
+                                                 List<String> assemblyTypes,
+                                                 DataSource dataSource) throws SQLException {
+        Map<String, Object> parameterValues = new HashMap<>();
+        parameterValues.put("assemblyType", assemblyTypes);
+        parameterValues.put("assembly", assembly);
+        parameterValues.put("ss_id", getSubsnpId(dbsnpRelease, batch, assembly, assemblyTypes, dataSource));
+        parameterValues.put("batch", batch);
+        return parameterValues;
+    }
+
+    private Long getSubsnpId(String dbsnpRelease, int batch, String assembly, List<String> assemblyTypes,
+                             DataSource dataSource) throws SQLException {
+        String joinedAssemblyTypes = assemblyTypes.stream().map(this::quote).collect(Collectors.joining(","));
```

Comment:
Since Java 8, you can use String.join() method for this:

https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-

Reply:
still, the main part here is `.map(this::quote)`